### PR TITLE
[DiD] S7-S9 revisions

### PR DESCRIPTION
**Pull request by nemaara**
As the title states, updates the dialogue in DiD S7.

### CODE REVIEW

**Comment by nemaara** [comment]
After some discussion with josteph, I've decided I'll be updating the gameplay for S7 and S8 simultaneously (they will depend on each other). So, this PR will no longer only be a dialogue revision; there will be gameplay changes in it as well.

**Comment by jostephd** [comment]
Labeled "in progress" until the S8 changes are pushed for review.

File: data/campaigns/Descent_Into_Darkness/scenarios/07_A_Small_Favor.cfg [hunk 1]
```diff
@@ -187,33 +234,155 @@
...
+            [/move_unit]
+            [move_unit]
+                id=Guard5
+                side=2
+                to_x,to_y=21,11
+            [/move_unit]
+            [move_unit]
+                id=Guard4
+                side=2
+                to_x,to_y=20,9
```
Comment (stevecotton) [inline data/campaigns/Descent_Into_Darkness/scenarios/07_A_Small_Favor.cfg]: The guards are shown as having only 3 mp, but on AI turn 9 Guard4 moves from 21,15 to 20,9, which should need 6 mp.

Comment (nemaara) [reply data/campaigns/Descent_Into_Darkness/scenarios/07_A_Small_Favor.cfg]: Should be okay now, I made some fixes.

**Comment by nemaara** [comment]
I've playtested both scenarios. They're meant to be balanced around getting 5-7 level 3 units (ghosts are best) and seem to be of reasonable difficulty with an army like that. I can provide a save if anyone else wants to playtest as well.

**Comment by jostephd** [comment]
How much gold+carryover is the player expected to start with?

**Comment by nemaara** [comment]
I assumed base gold, 120, which gives you 6 recalls. Some minor carryover will get you to 7; I passed on  5 without too much trouble. If you have more carryover, these scenarios will be easier, but I don't think it's a good idea to assume people will have any.

**Comment by nemaara** [comment]
Now includes S9 because balancing will depend heavily on the previous two scenarios.

File: data/campaigns/Descent_Into_Darkness/scenarios/07_A_Small_Favor.cfg [hunk 2]
```diff
@@ -451,7 +451,7 @@
 
         [message]
             speaker=Darken Volk
-            message= _ "The stolen book lies within Lord Karres’s manor at the north end of the city. It will be difficulty for us to defeat the entire city guard, so we must stay hidden as long as possible. For now, we can use this fog to our advantage. I will cast a spell using it to obscure the vision of the town guards. It should not affect us, however."
+            message= _ "The stolen book lies within Lord Karres’s manor at the north end of the city. It will be difficult for us to defeat the entire city guard, so we must stay hidden as long as possible. For now, we can use this fog to our advantage. I shall use it to cast a spell to obscure town guards’ vision. It should not affect us, however."
```
Comment (jostephd) [inline data/campaigns/Descent_Into_Darkness/scenarios/07_A_Small_Favor.cfg]: Missing "the" after "obscure".

**Pull request merged**